File: 001-two-sum/eder/solution_eder.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for (let i = 0; i < nums.length; i++) {
        let n = nums[i];
        
        // indexOf recebe um segundo parâmetro indicando a partir de onde buscar.
        // Usar i + 1 garante que não vamos usar o mesmo elemento duas vezes.
        let index_target = nums.indexOf(target - n, i + 1);
        
        // indexOf retorna -1 quando não encontra o elemento
        if (index_target !== -1) {
            return [i, index_target];
        }
    }
};

File: 001-two-sum/eder/solution_eder.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        for i, n in enumerate(nums):
            for i2, n2 in enumerate(nums):
                if i == i2:
                    continue
                if n + n2 == target:
                    return [i, i2]
